magento/magento2
# Fixed table difference insert during price reindex

### Description (*)
Problem appears when dimensions are not used.
`catalog_product_index_price` is now incompatible with `catalog_product_index_price_tmp` as new ID columnt added
https://devdocs.magento.com/guides/v2.4/release-notes/backward-incompatible-changes/reference.html
so method works incorrectly as final query contains additional column
```
\Magento\Catalog\Model\Indexer\Product\Price\AbstractAction::_syncData
```

We faced this issue while running graphql tests when indexers is set - `update on save`
The issue might be skipped in fixtures but when tests are placing the order - it triggers reindex that blocks order placing.

In query logs we may see how temp table is created
```
## 140 ## QUERY
SQL: CREATE TEMPORARY TABLE IF NOT EXISTS `catalog_product_index_price_temp` LIKE `catalog_product_index_price_tmp`
AFF: 0
TIME: 0.0008
```


### Related Pull Requests


### Fixed Issues (if relevant)

cannot reindex price 
```
Magento\Framework\Exception\LocalizedException: SQLSTATE[21S01]: Insert value list does not match column list: 1136 Column count doesn't match value count at row 1, query was: INSERT INTO `catalog_product_index_price` SELECT `ip_tmp`.* FROM `catalog_product_index_price_temp` AS `ip_tmp` ON DUPLICATE KEY UPDATE `tax_class_id` = VALUES(`tax_class_id`), `price` = VALUES(`price`), `final_price` = VALUES(`final_price`), `min_price` = VALUES(`min_price`), `max_price` = VALUES(`max_price`), `tier_price` = VALUES(`tier_price`) in /home/www/magento/vendor/magento/module-catalog/Model/Indexer/Product/Price/Action/Row.php on line 32
```
1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

0. Set indexer update on save
1. Create product on 2.4.5 installation
2. Create custom script to reindex
3. RUN `Magento\Catalog\Model\Indexer\Product\Price::executeRow`
4. See error.
5. Apply patch from this PR
6. retry reindex - no error

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36471: Fixed table difference insert during price reindex
